tests/test_switch_to_branch.py

```
@@ -0,0 +1,59 @@
+    res = asyncio.run(switch_to_branch("dev", clone))
+    cur = subprocess.run(["git", "rev-parse", "--abbrev-ref", "HEAD"], cwd=clone, capture_output=True, text=True).stdout.strip()
+    assert cur == "dev"
+    assert (clone / "d").exists()
+    assert res.new_sha == _sha(clone)
+
+
+def test_stashes_dirty_tree_and_tags_recovery(remote_and_clone):
+    clone = remote_and_clone
+    (clone / "f").write_text("dirty-edit")
+    res = asyncio.run(switch_to_branch("dev", clone))
+    cur = subprocess.run(["git", "rev-parse", "--abbrev-ref", "HEAD"], cwd=clone, capture_output=True, text=True).stdout.strip()
```

Ruff (0.15.15)

[error] Starting a process with a partial executable path

(S607)